### PR TITLE
Add Wizard (new component)

### PR DESCRIPTION
**Pull request by M-i-k-e-l**

### CODE REVIEW

File: src/components/wizard/WizardStep.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+import {StyleSheet} from 'react-native';
+import View from '../view';
+import Text from '../text';
+import Image from '../image';
+import TouchableOpacity from '../touchableOpacity';
+import {PureBaseComponent} from '../../commons';
+import Colors from '../../style/colors';
+import BorderRadiuses from '../../style/borderRadiuses';
+import Spacings from '../../style/spacings';
+import {States, StatesConfig} from './WizardStates';
```
Comment (ethanshar) [inline src/components/wizard/WizardStep.js]: States is not being used, it will fail lint

File: src/components/wizard/WizardStep.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    } else {
+      return config[propName];
+    }
+  }
+
+  renderCircle(config, activeConfig) {
+    const {testID, index, onPress, indexLabelStyle} = this.getThemeProps();
+    const hitSlopSize = Spacings.s2;
+    const color = this.getProp(activeConfig, config, 'color');
+    const circleColor = this.getProp(activeConfig, config, 'circleColor');
```
Comment (ethanshar) [inline src/components/wizard/WizardStep.js]: I would use the `color` value as default for the `circleColor`.

File: src/components/wizard/WizardStep.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+    return (
+      <TouchableOpacity
+        testID={`${testID}.circle`}
+        style={[styles.circle, {borderColor: circleColor}]}
+        onPress={enabled ? onPress : undefined}
+        hitSlop={{top: hitSlopSize, bottom: hitSlopSize, left: hitSlopSize, right: hitSlopSize}}
+        disabled={!enabled}
+      >
+        {_.isUndefined(icon) ? (
```
Comment (ethanshar) [inline src/components/wizard/WizardStep.js]: this is one of the cases that checking with `_.isUndefined` is redundant. `icon ? ... : ....` is enough

File: src/components/wizard/WizardStep.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+          <Image testID={`${testID}.image`} source={icon} tintColor={color}/>
+        )}
+      </TouchableOpacity>
+    );
+  }
+
+  render() {
+    const {
+      testID,
+      state,
```
Comment (ethanshar) [inline src/components/wizard/WizardStep.js]: `state` is missing from prop types

File: src/components/wizard/WizardStep.js [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+     */
+    indexLabelStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
+    /**
+     * Whether the step should be enabled
+     */
+    enabled: PropTypes.bool
+  };
+
+  getProp(activeConfig, config, propName) {
+    const props = this.getThemeProps();
```
Comment (ethanshar) [inline src/components/wizard/WizardStep.js]: why extracting the props like that?
also `index` and `activeIndex` can be taken directly from `this.props`

Comment (ethanshar) [reply src/components/wizard/WizardStep.js]: ok I see why u did that. 

anyway, you can shorten the way you get the config values. 
the way I see it there are 3 sets of config. 
the state config, the active config and the props (which should are identical to the config structure)

you can create some sort of merged config out of those three and just take the final values from the merge config. 
it should turn this whole code two 2-3 lines of code (:

File: src/components/wizard/WizardStep.js (hunk 3 above)
Comment (M-i-k-e-l) [reply src/components/wizard/WizardStep.js]: I was unsure if it can be 0

File: src/components/wizard/WizardStep.js (hunk 4 above)
Comment (M-i-k-e-l) [reply src/components/wizard/WizardStep.js]: Done.

File: src/components/wizard/WizardStep.js (hunk 1 above)
Comment (M-i-k-e-l) [reply src/components/wizard/WizardStep.js]: Added state to props, it is now being used

File: src/components/wizard/WizardStep.js (hunk 5 above)
Comment (M-i-k-e-l) [reply src/components/wizard/WizardStep.js]: Done.

File: src/components/wizard/WizardStep.js (hunk 2 above)
Comment (M-i-k-e-l) [reply src/components/wizard/WizardStep.js]: Done

**Pull request merged**